rust/sensord/src/controller/context.rs

```diff
@@ -0,0 +1,206 @@
+    marker: std::marker::PhantomData<S>,
+}
+
+impl<S> ContextBuilder<S, NoScanner> {
+    /// Creates a new context builder.
+    pub fn new() -> Self {
+        Self {
+            scanner: NoScanner,
+            result_config: None,
+            feed_config: None,
+            api_key: None,
+            marker: std::marker::PhantomData,
+            enable_get_scans: false,
+        }
+    }
+}
+
+impl<S, T> ContextBuilder<S, T> {
+    /// Sets the result config.
+    pub fn result_config(mut self, config: impl Into<ResultContext>) -> Self {
+        self.result_config = Some(config.into());
+        self
+    }
+
+    /// Sets the feed config.
+    pub fn feed_config(mut self, config: impl Into<FeedContext>) -> Self {
+        self.feed_config = Some(config.into());
+        self
+    }
+
+    /// Sets the api key.
+    pub fn api_key(mut self, api_key: impl Into<Option<String>>) -> Self {
+        self.api_key = api_key.into();
+        self
+    }
+
+    /// Enables the GET /scans endpoint.
+    pub fn enable_get_scans(mut self, enable: bool) -> Self {
+        self.enable_get_scans = enable;
+        self
+    }
+}
+
+impl<S> ContextBuilder<S, NoScanner>
+where
+    S: Clone,
+{
+    /// Sets the scanner. This is required.
+    pub fn scanner(self, scanner: S) -> ContextBuilder<S, Scanner<S>>
+    where
+        S: super::Scanner + 'static + std::marker::Send + std::marker::Sync + std::fmt::Debug,
+    {
+        let Self {
+            result_config,
+            feed_config,
+            api_key,
+            enable_get_scans,
+            scanner: _,
+            marker: _,
+        } = self;
+        ContextBuilder {
+            scanner: Scanner(scanner),
+            result_config,
+            feed_config,
+            marker: std::marker::PhantomData,
+            api_key,
+            enable_get_scans,
+        }
+    }
+}
+
+impl<S> ContextBuilder<S, Scanner<S>> {
+    pub fn build(self) -> Context<S> {
+        Context {
+            scanner: self.scanner.0,
+            response: Default::default(),
+            scans: Default::default(),
+            oids: Default::default(),
+            result_config: self.result_config,
+            feed_config: self.feed_config,
+            abort: Default::default(),
+            api_key: self.api_key,
+            enable_get_scans: self.enable_get_scans,
+        }
+    }
+}
+
+#[derive(Debug)]
+/// The context of the application
+pub struct Context<S> {
+    /// The scanner that is used to start, stop and fetch results of scans.
+    pub scanner: S,
+    /// Creates responses
+    pub response: response::Response,
+    /// The scans that are being tracked.
+    ///
+    /// It is locked to allow concurrent access, usually the results are updated
+    /// with a background task and appended to the progress of the scan.
+    pub scans: RwLock<HashMap<String, crate::scan::Progress>>,
+    /// The OIDs thate can be handled by this sensor.
+    pub oids: RwLock<(String, Vec<String>)>,
```

Comment:
In this implementation we respond with `["feed_hash",["oid1","oid2",...]]` on `/vts`.
This is different from the documented API, which does not include a hash value, but just a list of OIDs.
Either adjust the implementation or the openAPI specification.